### ISSUE TITLE
[ui5-carousel] [accessibility] (Role with implied hidden content has keyboard focus)

### ISSUE THREAD

**Issue by dmitrykostenko** [SAP/ui5-webcomponents#10198]
### Bug Description

 When attempting to include focusable content such as links, buttons, or any interactive elements inside a carousel, an accessibility error occurs. 
 
Accessibility issue:
**Role with implied hidden content has keyboard focus**

Some elements have a role that implies their internal structure should be hidden from assistive technologies. In other words, their sub-elements have a presentational role.

If a sub-element has keyboard focus, it should be moved outside of the presentation role.
![image](https://github.com/user-attachments/assets/e98f9549-37ad-4454-9374-e03414956111)

![image](https://github.com/user-attachments/assets/b940394a-2f81-454a-b1b9-462eff492b36)

### Steps to Reproduce

add focusable element inside carousel
![image](https://github.com/user-attachments/assets/2da0e643-0338-4dcf-a177-14ad491251f8)

### Priority

Medium

### UI5 Web Components Version

1.24.12

### Browser

Chrome

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by dobrinyonkov** [comment]
Hello @dmitrykostenko,

Could you please provide more details about the issue you are experiencing? It would be especially helpful if you could share a code snippet or a live example that demonstrates the problem. This will help us better understand the issue and offer an appropriate solution.

Which tool are you using to check accessibility? Additionally, could you clarify the user impact of this issue?

I attempted to reproduce the problem in our [playground sample](https://sap.github.io/ui5-webcomponents/play/#eyJpbmRleC5odG1sIjp7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhLS0gcGxheWdyb3VuZC1mb2xkIC0tPlxuPCFET0NUWVBFIGh0bWw-XG48aHRtbCBsYW5nPVwiZW5cIj5cblxuXG5cbjxoZWFkPlxuICAgIFxuICAgIDxzdHlsZT5cbiAgICAgICo6bm90KDpkZWZpbmVkKSB7XG4gICAgICAgIGRpc3BsYXk6IG5vbmU7XG4gICAgICB9XG5cbiAgICBodG1sIHtcbiAgICAgIGZvcmNlZC1jb2xvci1hZGp1c3Q6IG5vbmU7XG4gICAgfVxuICAgIDwvc3R5bGU-XG5cbiAgICBcbiAgICA8c3R5bGU-XG4gICAgICAqOm5vdCg6ZGVmaW5lZCkge1xuICAgICAgICBkaXNwbGF5OiBub25lO1xuICAgICAgfVxuXG4gICAgaHRtbCB7XG4gICAgICBmb3JjZWQtY29sb3ItYWRqdXN0OiBub25lO1xuICAgIH1cbiAgICA8L3N0eWxlPlxuXG4gICAgPG1ldGEgY2hhcnNldD1cIlVURi04XCI-XG4gICAgPG1ldGEgbmFtZT1cInZpZXdwb3J0XCIgY29udGVudD1cIndpZHRoPWRldmljZS13aWR0aCwgaW5pdGlhbC1zY2FsZT0xLjBcIj5cbiAgICA8dGl0bGU-U2FtcGxlPC90aXRsZT5cbiAgICA8bGluayByZWw9XCJzdHlsZXNoZWV0XCIgaHJlZj1cIi4vbWFpbi5jc3NcIj5cbjwvaGVhZD5cblxuPGJvZHkgc3R5bGU9XCJiYWNrZ3JvdW5kLWNvbG9yOiB2YXIoLS1zYXBCYWNrZ3JvdW5kQ29sb3IpXCI-XG4gICAgPCEtLSBwbGF5Z3JvdW5kLWZvbGQtZW5kIC0tPlxuXG4gICAgPHVpNS1jYXJvdXNlbD5cbiAgICAgICAgPGltZyBzcmM9XCJodHRwczovL3NhcC5naXRodWIuaW8vdWk1LXdlYmNvbXBvbmVudHMvaW1hZ2VzL3NhbXBsZTEuanBnXCIgYWx0PVwiTGFuZHNjYXBlIDFcIj5cbiAgICAgICAgPGltZyBzcmM9XCJodHRwczovL3NhcC5naXRodWIuaW8vdWk1LXdlYmNvbXBvbmVudHMvaW1hZ2VzL3NhbXBsZTIuanBnXCIgYWx0PVwiTGFuZHNjYXBlIDJcIj5cbjxkaXY-XG4gICAgICAgIDxpbWcgc3JjPVwiaHR0cHM6Ly9zYXAuZ2l0aHViLmlvL3VpNS13ZWJjb21wb25lbnRzL2ltYWdlcy9zYW1wbGUzLmpwZ1wiIGFsdD1cIkJ1bGJcIj5cbiAgICAgICAgPHVpNS1saW5rPnRlc3Q8dWk1LWxpbms-XG4gICAgICAgICAgPC9kaXY-XG4gICAgPC91aTUtY2Fyb3VzZWw-XG4gICAgPCEtLSBwbGF5Z3JvdW5kLWZvbGQgLS0-XG4gICAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwibWFpbi5qc1wiPjwvc2NyaXB0PlxuPC9ib2R5PlxuXG48L2h0bWw-XG48IS0tIHBsYXlncm91bmQtZm9sZC1lbmQgLS0-XG4ifSwibWFpbi5qcyI6eyJuYW1lIjoibWFpbi5qcyIsImNvbnRlbnQiOiIvKiBwbGF5Z3JvdW5kLWhpZGUgKi9cbmltcG9ydCBcIi4vcGxheWdyb3VuZC1zdXBwb3J0LmpzXCI7XG4vKiBwbGF5Z3JvdW5kLWhpZGUtZW5kICovXG5pbXBvcnQgXCJAdWk1L3dlYmNvbXBvbmVudHMvZGlzdC9DYXJvdXNlbC5qc1wiO1xuaW1wb3J0IFwiQHVpNS93ZWJjb21wb25lbnRzL2Rpc3QvTGluay5qc1wiOyJ9LCJtYWluLmNzcyI6eyJuYW1lIjoibWFpbi5jc3MiLCJjb250ZW50IjoiaW1nIHtcbiAgICBtYXgtaGVpZ2h0OiAzMHJlbTtcbn0ifX0), but the issue was reported using the AMP tool.

Looking forward to your input!

Best regards,
Dobrin

**Comment by TeodorTaushanov** [comment]
When testing the following code with Axe DevTool
```
 <ui5-carousel>
          <div><a href="#">Link</a></div>
          <img src="https://sap.github.io/ui5-webcomponents/v1/images/sample2.jpg" alt="Landscape 2" />
          <img src="https://sap.github.io/ui5-webcomponents/v1/images/sample3.jpg" alt="Bulb" />
</ui5-carousel>
```

an error is flagged indicating that interactive controls must not be nested.

This issue arises because the carousel uses "listbox" with "option" roles instead of "list" with "listitem" roles. We should investigate whether the role assignments should be adjusted.

**Comment by GerganaKremenska** [comment]
Hello @dmitrykostenko,

This issue is hanled via this JIRA: BGSOFUIRODOPI-3382, and it is already a duplicated on this reported [issue](https://github.com/SAP/ui5-webcomponents/issues/10198).

So i am closing it and you can follow its progress in the linked issue.

Best Regards,
Gergana
